### ISSUE TITLE
PostgreSQL 10 compatibility

### ISSUE THREAD

**Issue by alexeyklyukin** [psycopg/psycopg2#489]
When running on PostgreSQL 10devel, psycopg2 complains about inability to determine PostgreSQL version.

I think the problem lies in the following regexp

    verre = re.compile(
        r"(\d+)\.(\d+)(?:(?:\.(\d+))|(devel|(alpha|beta|rc)\d+))")

I did a quick and dirty patch with
```diff
diff --git a/setup.py b/setup.py
index 3f02183..eaa6a33 100644
--- a/setup.py
+++ b/setup.py
@@ -405,14 +405,19 @@ class psycopg_build_ext(build_ext):
                 pgversion = pg_config_helper.query("version").split()[1]
             except:
                 pgversion = "7.4.0"
-
-            verre = re.compile(
-                r"(\d+)\.(\d+)(?:(?:\.(\d+))|(devel|(alpha|beta|rc)\d+))")
+            if not pgversion[:2].isdigit():
+                verre = re.compile(
+                    r"(\d+)\.(\d+)(?:(?:\.(\d+))|(devel|(alpha|beta|rc)\d+))")
+            else:
+                verre = re.compile(
+                   r"(\d+)(?:(?:\.(\d+))|(devel|(alpha|beta|rc)\d+))")
             m = verre.match(pgversion)
             if m:
                 pgmajor, pgminor, pgpatch = m.group(1, 2, 3)
                 if pgpatch is None or not pgpatch.isdigit():
                     pgpatch = 0
+                if pgminor is None or not pgminor.isdigit():
+                    pgminor = 0
                 pgmajor = int(pgmajor)
                 pgminor = int(pgminor)
                 pgpatch = int(pgpatch)
```
but I think the original regexp is wrong, since for the "devel" version it gives me an extra None group, i.e. for the "9.2devel" I'm getting ('9', '2', None, 'devel', None) instead of ('9', '2', 'devel',...).

**Comment by fogzot** [comment]
The regex is fine, just returns `None` for version parts that are not present in the version string.
The correct fix is to make the second group optional:

    (\d+)(?:\.(\d+))?(?:(?:\.(\d+))|(devel|(alpha|beta|rc)\d+))

**Comment by fogzot** [comment]
Will be fixed by pulling #504.
